tests/unit/test_ast.c

```diff
@@ -0,0 +1,73 @@
+#include <stdio.h>
+#include <string.h>
+#include "assert.h"
+#include "../../src/parser/ast.h"
+#include "../../src/query_executor.h"
+
+void test_validate_set_clause() {
+    char *reason = malloc(sizeof(char) * 30);
+    char *errMsg = NULL;
+
+    // Valid query
+    char *query = "MATCH (n { name: 'Andres' }) SET n.surname = 'Taylor'";
+    AST_QueryExpressionNode* ast = ParseQuery(query, strlen(query), &errMsg);
+    assert(ast != NULL);
+    assert(_Validate_SET_Clause(ast, &reason) == AST_VALID);
+
+    // // invalid query
+    query = "MATCH (n { name: 'Andres' }) SET n.surname = 'Taylor', m.middle_name = '23'";
+    ast = ParseQuery(query, strlen(query), &errMsg);
+    assert(ast != NULL);
+    assert(_Validate_SET_Clause(ast, &reason) == AST_INVALID);
+    assert(strcmp(reason, "m not defined") == 0);
+
+    free(reason);
+}
+
+void test_validate_delete_clause() {
+    char *reason = malloc(sizeof(char) * 30);
+    char *errMsg = NULL;
+
+    // Valid query
+    char *query = "MATCH (n { name: 'Andres' }) DELETE n";
+    AST_QueryExpressionNode* ast = ParseQuery(query, strlen(query), &errMsg);
+    assert(ast != NULL);
+    assert(_Validate_DELETE_Clause(ast, &reason) == AST_VALID);
+
+    // invalid query
+    query = "MATCH (n { name: 'Andres' }) DELETE n, m";
+    ast = ParseQuery(query, strlen(query), &errMsg);
+    assert(ast != NULL);
+    assert(_Validate_DELETE_Clause(ast, &reason) == AST_INVALID);
+    assert(strcmp(reason, "m not defined") == 0);
+
+    free(reason);
+}
+
+void test_validate_return_clause() {
+    char *reason = malloc(sizeof(char) * 30);
+    char *errMsg = NULL;
+
+    // Valid query
+    char *query = "MATCH (n { name: 'Andres' }) RETURN n";
+    AST_QueryExpressionNode* ast = ParseQuery(query, strlen(query), &errMsg);
+    assert(ast != NULL);
+    assert(_Validate_RETURN_Clause(ast, &reason) == AST_VALID);
+
```

Comment:
There's no check here which really traverse an expression tree,
Example of a "deep" expression tree:
return AVG( 2 / ABS(n.age) )